### PR TITLE
egress_ip: Move functions to egressIPController

### PR DESCRIPTION
**Pull request by dave-tucker**
**- What this PR does and why is it needed**

This commit formalizes egressIPController as the place where all
egressIP related functions live.

It moves a number of egressIP related functions out of the OvnController
as they only require access to the WatchFactory - I'll be working on
factoring this out in future.

When moving these functions over I noticed that EgressIP was fetching from
the API Server and not from the cache. Once I'd fixed this to use the
informer cache there was only UpdateEgressIP left in the Kube interface.
As it's only used in the egress IP code, I went ahead and in-lined it in
egress_ip.go to save passing the Kube interface around everywhere.

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/factory/factory.go
Comment (alexanderConstantinescu) [inline go-controller/pkg/factory/factory.go:506]: `GetPods returns all the pods in a given namespace` -> `GetPodsWithSelector returns a list of pods matching the selector`

**Review by alexanderConstantinescu — CHANGES_REQUESTED** [review]
Looks good! Thanks for the help on this @dave-tucker . I just had one small nit and one personal request: that we merge this after #1784 . I will need to back-port that PR to OpenShift 4.6 and I would like to avoid increasing the back-port diff (which would happen if this merges first and #1784 rebases on this). Is that fine for you? It should merge pretty soon (today/tomorrow) and it shouldn't be difficult for you to rebase on.

File: go-controller/pkg/factory/factory.go
Comment (dave-tucker) [reply go-controller/pkg/factory/factory.go]: Fixed, Thanks!

**Comment by dave-tucker** [comment]
@alexanderConstantinescu Thanks for reviewing. I'm fine to rebase this after #1784 merges!

File: go-controller/pkg/ovn/egressip.go
Comment (alexanderConstantinescu) [inline go-controller/pkg/ovn/egressip.go:464]: Small question: an object not found when calling `GetEgressIP` would mean that the informer looked in its cache, did not find it and then queried the API server for it and did not find it there either, correct? Wouldn't it be more correct to leave the old log message as it was? If that is not the case: then why wasn't L459 changed as well?

**Pull request closed without merging**
